### ISSUE TITLE
Create Figma wireframes for the plan builder

### ISSUE THREAD

**Issue by mariapraetzellis** [CDLUC3/dmsp_frontend_prototype#32]

**Comment by mariapraetzellis** [comment]
Notes from team meeting: 

Uploaded plans will be integrated into the full plan builder.

Writing plans via the DMP Tool is the priority workflow rather than uploading. 

Decided not to move forward with the ability to modify dates/other information from the downloaded plan. We'll revist this later if needed.

**Comment by fraserclark** [comment]
Hey All,

We have completed the wireframes for the Plan Creator and flow

**Video Walkthrough**
https://www.loom.com/share/ae6593f32d214353938cadeab06627c2?sid=04d19283-b8b4-4294-bc94-864b79cd221d

**Prototypes**
The create flow leads to the project flow but to make it easy, I have made a second "entry point" - so you can jump into project 
You can press "R" to restart to beginning at anytime.


**Create flow** 
https://www.figma.com/proto/RRtoRLyr5Spf0KiyRfEclt/DMP-Explain?node-id=177-626&t=pGJl655GG7A2dvdJ-0&scaling=min-zoom&page-id=177%3A504&starting-point-node-id=177%3A626&show-proto-sidebar=1

**Project Create/Edit**
https://www.figma.com/proto/RRtoRLyr5Spf0KiyRfEclt/DMP-Explain?node-id=177-642&t=pGJl655GG7A2dvdJ-0&scaling=min-zoom&page-id=177%3A504&starting-point-node-id=177%3A642&show-proto-sidebar=1

**Misc.**

- Research outputs - I'm working on this based on phase 1
- Related works - I'm working on this based on Phase 1 also - I wonder if there was any user research results which suggests we need to re-think or adjust this?
- Mock Project - Right now we don't have "Mock projects" in - we wonder if this is requirement or "draft" can effectively work as a way to "mock a project" etc



We went back and forward on the Requirements vs Guidance vs "Best Practice"- We looked at a lot of the real guidance so we tried to show "realistic-ish" amount of text and there is a lot of variations - we have tried to balance out not wanting to overwhelm the users on first "open", but also we tried to avoid hiding it away behind accordions etc

Let us know your general thoughts, and we can think about how we might make separate cards for screens or collection of screens for more feedback.

**Comment by cazinc** [comment]
Thanks @fraserclark 

As we all go through this to review and provide feedback, how do you want us to structure that? I see you've added a unique name to each slide in the wireframes in the yellow notes, like "Flow create plan", so can I suggest we just group our thoughts our questions by these here on this card, to start? We can always expand to separate cards if things need a bit more room for discussion? Does that work Ok for everyone else?

**Comment by fraserclark** [comment]
Yep, open to whatever works best for people. I can see  the value in splitting off the big topics into separate cards where required.

Appreciate there are a ton of screens to review and comment on

**Comment by briri** [comment]
Thanks @fraserclark and @cazinc The video was super helpful and I think things are looking great!

I have one comment about the overall flow. You may already have already catalogued this and just not covered in the video (or I missed it). Like phase 1, our ability to provide the 'Project search' is limited to only a handful of funders. We will need to retain that in the new system. 

We have not yet had any feedback on the related works section from Phase 1.

As for Mock projects, I'm not sure. I think we could possibly leave out now that we have introduced new language like 'Draft' and 'Publish'

**Comment by mariapraetzellis** [comment]
@fraserclark I'm really happy with how things are looking and how fast you could turn this around! 

This set covers a lot, so we will need more time for a comprehensive review. I will write out all of my comments on one card and group them by slide name. As @cazinc noted, from there, I think we can consider breaking them off into separate cards.

**Comment by fraserclark** [comment]
Hey @briri 

> I have one comment about the overall flow. You may already have already catalogued this and just not covered in the video (or I missed it). Like phase 1, our ability to provide the 'Project search' is limited to only a handful of funders. We will need to retain that in the new system.

**Project Search**
Hopefully I didn't accidentally skip it in the video - but it does exist in the project "create flow" - this should be a direct link to it.

https://www.figma.com/proto/RRtoRLyr5Spf0KiyRfEclt/DMP-Explain?node-id=177-562&t=pGJl655GG7A2dvdJ-0&scaling=min-zoom&page-id=177%3A504&starting-point-node-id=177%3A626

However, you are right, the search is missing at the point of "editing a project" - its only available on that initial "create" which is of course wrong - I will update this.

**Comment by cazinc** [comment]
Hi @briri @mariapraetzellis,

We've updated the Plan Builder wireframes now to reflect the requirements around [Collaborators/Contributors/PIs/**comment-only** sharing](https://docs.google.com/document/d/1uBryDvxlgyKGrHI4wqbGoiLqQ69jHoiL3TtJBUY8rO0/edit#heading=h.7io6sqbv21y5). 

This has been done in 3 places:
1) On the [Project Plan Overview](https://www.figma.com/proto/RRtoRLyr5Spf0KiyRfEclt/DMP-Explain?node-id=177-642&t=BWmBBuluZIM4zJsL-1) - you can now see a role for "Comment-only"
2) If you click **Edit Details** under Collaborators from the above, you'll see the Comment-only role shown here too
3) If you skip to (Add/Edit Collaborator)[https://www.figma.com/proto/RRtoRLyr5Spf0KiyRfEclt/DMP-Explain?node-id=260-583&t=BWmBBuluZIM4zJsL-1] you'll see a new "Comment-only access" option. It's shown bold here to differentiate it from the other options, we'd give this a bit more thought during the design phase. I think it perhaps needs a "What does this mean?" sort of link. 

How does this look? I think we'd already merged the concepts of adding PIs into a higher level "Add Collaborators" flow; this really just introduces the option to set a collaborator as comment-only. If memory serves this would be the way for a Researcher to request input not only from a peer but also from a Data Librarian, for example?

**Comment by mariapraetzellis** [comment]
Hi @cazinc, Thanks for the updates. A few comments on Collaborators 

- Can we make it more clear that in order to share a plan, users have to be listed as collaborators? 
- We'll need an additional role for "Other." 
- Add an ORCID ID look up on the Edit Collaborator page.[ DMP OPIoR](https://dmp.opidor.fr/) (a French version of DMP Tool) has a nice integration with ORCID that we could emulate. 
<img width="702" alt="Screen Shot 2024-06-18 at 7 41 59 AM" src="https://github.com/CDLUC3/dmsp_frontend_prototype/assets/27511575/2f8736a8-3236-489b-8d63-89a2aba28453">

**Comment by cazinc** [comment]
Hi @mariapraetzellis thanks, we'll start to make some edits to the wireframes. 

On the "other" role - I worry this won't be clear to users, specifically that the implications for edit rights/comment rights/no rights won't be clear. Is there some other term like "View-only" that matches the requirement?

**Comment by cazinc** [comment]
Hi @mariapraetzellis we have a few more edits to this ready for review:

- If you go to (this screen)[https://www.figma.com/proto/RRtoRLyr5Spf0KiyRfEclt/DMP-Explain?node-id=183-522&t=YbPBZMEafj1T2zib-1] you'll see the ORCiD lookup for Collaborators; I think it maybe just hard to spot in the "prototype spaghetti". Do you want it modified to also work withy ROR IDs per the OPIoR example? 
- I've made several changes related to your first bullet point about clarifying who can feedback:
  - New "Can Feedback", "Can Edit" role grid on the [Edit Collaborators screen](https://www.figma.com/proto/RRtoRLyr5Spf0KiyRfEclt/DMP-Explain?node-id=260-583&t=YbPBZMEafj1T2zib-1) to clarify who can do what. (See my note above about "Other" too - how would we define it here?)
  - New **Request feedback** links from the [Project Overview screen](https://www.figma.com/proto/RRtoRLyr5Spf0KiyRfEclt/DMP-Explain?node-id=177-642&t=YbPBZMEafj1T2zib-1) on each Plan. Follow these to see a new process where Researchers can request feedback. This seemed to make more sense being _plan_-related rather than _project_-related I think? It needs fleshed out - including the whole feedback process itself - but I wanted to check this looks good to you before going too far.

**Comment by cazinc** [comment]
Hi, I have several updates to the Plan Builder ready for review. These are related to Thursday's discussion about sharing versus feedback versus project roles. 

1. The [**Collaborators** screen](https://www.figma.com/proto/RRtoRLyr5Spf0KiyRfEclt/DMP-Explain?node-id=260-583&t=txnTCBiWD5IwsGUu-1) has been changed back to show just project roles (and I've renamed the title for this section "Project role(s)" to be more clear. The misplaced Feedback item has been removed. 
2. The new [**Share** screen](https://www.figma.com/proto/RRtoRLyr5Spf0KiyRfEclt/DMP-Explain?node-id=396-932&t=txnTCBiWD5IwsGUu-1) has been reworked to just about sharing (and no more mention of feedback. This is using a slightly modified version of the same ORCiD search tool, as this seemed like it might be sensible? You'll see there's also an indication of how the Researcher might share with someone who's not a DMP Tool user. 
3. The **Request feedback** button added to each Plan has been retermed "Share" as requested. 

There is no longer any explicit "request feedback" option but from what was discussed in the meeting, I think this is right? I am wondering though if we need to think when Feedback is requested, even if it is automated? Does it happen when the researcher Publishes the Plan (too late?) or if there is some intermediate stage before publishing that makes it clear you are ready for (required) feedback from (a non editable list of org admins/data librarians/whomever)?
